Plugins/Public/balancemagic/Main.cpp

```diff
@@ -455,23 +487,151 @@ void __stdcall HkCb_AddDmgEntry(DamageList *dmg, unsigned short p1, float damage
 	}
 }
 
+void __stdcall FireWeapon(unsigned int iClientID, struct XFireWeaponInfo const &wpn)
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	if (!iLoadedBurstFires)
+		return;
+
+	IObjInspectImpl *obj = HkGetInspect(iClientID);
+	if (obj)
+	{
+		CShip* cship = (CShip*)HkGetEqObjFromObjRW((IObjRW*)obj);
+
+		for (short* slot = wpn.sArray1; slot != wpn.sArray2; slot++)
+		{
+			CEquipManager* eqmanager = (CEquipManager*)((char*)cship + 0xE4);
+			CEquip *equip = eqmanager->FindByID(*slot);
+			if (CEGun::cast(equip))
+			{
+				CEGun* cgun = (CEGun*)equip;
+				map<unsigned int, struct AmmoGenerator>::iterator iter = mapBurstFire.find(cgun->GunArch()->iArchID);
+				if (iter != mapBurstFire.end())
+				{
+					aClientData[iClientID].mapBurstFills[*slot].ammo = iter->second.ammo;
+					aClientData[iClientID].mapBurstFills[*slot].count++;
+					aClientData[iClientID].mapBurstFills[*slot].timestamp = GetTimeInMS() + iter->second.timestamp;
+					ConPrint(L"BURSTFIRE: Client %u fired a round of 0x%08X from slot %u.\n", iClientID, iter->second.ammo, *slot);
```

Comment:
is this something we want to print potentially very often?

Reply:
Yeah this is in my testing bits but the whole thing is dummied out at the moment so it doesn't matter